test/samlp.tests.js

```diff
@@ -541,6 +488,79 @@ describe('samlp (unit tests)', function () {
         done();
       });
     });
+
+    it('should use providername option', function(done) {
+      var samlp = new Samlp({
+        identityProviderUrl: server.identityProviderUrl,
+        requestTemplate: '<samlp:AuthnRequest xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol" ID="@@ID@@" IssueInstant="@@IssueInstant@@" ProviderName="@@ProviderName@@" ProtocolBinding="@@ProtocolBinding@@" Version="2.0"><saml:Issuer xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion">@@Issuer@@</saml:Issuer></samlp:AuthnRequest>',
+        protocol: 'samlp',
+        providerName: 'Some name'
+      });
+      samlp.getSamlRequestParams({}, function(err, params) {
+        expect(params).to.have.property('SAMLRequest');
+        var SAMLRequest = params.SAMLRequest;
+
+        zlib.inflateRaw(new Buffer(SAMLRequest, 'base64'), function (err, buffer) {
+          if (err) return done(err);
+
+          var request = buffer.toString();
+          var doc = new xmldom.DOMParser().parseFromString(request);
+
+          expect(doc.documentElement.getAttribute('ProviderName'))
+            .to.equal('Some name');
+
+          done();
+        });
+      });
+    });
+
+    describe('signing', function () {
+      describe('HTTP-POST or HTTP-Redirect without deflate encoding', function () {
+        it('should error if the requestTemplate is malformed', function (done) {
+          var options = {
+            protocolBinding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST',
+            signingKey: {
+              key: fs.readFileSync(__dirname + '/test-auth0.key'),
+              cert: fs.readFileSync(__dirname + '/test-auth0.pem')
+            },
+            identityProviderUrl: 'https://example.com?RelayState=foo',
+            requestTemplate: '<samlp:AuthnRequest attribute="></samlp:AuthnRequest>'
+          };
+          this.samlp.getSamlRequestParams(options, function(err, result) {
+            expect(err).to.be.an.Error;
+            expect(err.message).to.equal('end tag name: samlp:AuthnRequest is not match the current start tagName:undefined');
+            expect(result).to.not.exist;
+            done();
+          });
+        });
+
+        it('should place the signature after the issuer', function(done){
```

Comment:
And this was also moved unchanged. I was going to implement a new "happy" test, but then I found it was actually available, but in the wrong place.